File: Data Structures & Algorithms/maximum-twin-sum-of-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        stack = []
        
        while head:
            stack.append(head.val)
            head = head.next
        
        left = 0
        right = len(stack) - 1

        max = 0

        while left < right:
            sum = stack[left] + stack[right]
            if sum > max:
                max = sum
            left += 1
            right -= 1
        
        return max
            
        

File: Data Structures & Algorithms/reverse-string/submission-0.java
class Solution {
    public void reverseString(char[] s) {
        int left = 0;
        int right = s.length -1;

        while (left < right) {
            char l = s[left];
            char r = s[right];

            s[left] = r;
            s[right] = l;
            
            left++;
            right--;
        }
    }
}